### PR TITLE
Backward compatibility

### PR DESCRIPTION
**Pull request by mohithapani**
Fixing the backward incompatibility in Here Account class

### CODE REVIEW

File: here-oauth-client/src/main/java/com/here/account/oauth2/HereAccount.java [hunk 1]
```diff
@@ -133,6 +133,41 @@
      * This class cannot be instantiated.
      */
     private HereAccount() {}
+    
+    
+    
+    
+    /**
+     * 
+     * THIS METHOD SHOULD NOT BE USED.
```
Comment (kenmccracken) [inline here-oauth-client/src/main/java/com/here/account/oauth2/HereAccount.java]: Given from this comment you don't want the developer to use this method any more, also mark the method '@deprecated' / '@Deprecated'.

Don't use all caps.  Just say 
```
Use of HereAccessTokenProvider.builder().build() is preferred going forward, as this 
simplifies integration.  Use of getTokenEndpoint(HttpProvider httpProvider, 
ClientAuthorizationRequestProvider clientAuthorizationRequestProvider) is also okay.
```

File: here-oauth-client/src/test/java/com/here/account/oauth2/HereAccountTest.java [hunk 2]
```diff
@@ -715,4 +715,22 @@ public InputStream getResponseBody() throws IOException {
...
+    @Test(expected = NullPointerException.class)
+    public void test_getTokenEndpoint_old_null_url() {
+        HttpProvider mockHttpProvider = Mockito.mock(HttpProvider.class);
+        ClientCredentialsProvider mockClientCredentialsProvider =
+                Mockito.mock(ClientCredentialsProvider.class);
+        Mockito.doReturn(null)
+                .when(mockClientCredentialsProvider).getTokenEndpointUrl();
+        Mockito.doReturn(HttpConstants.HttpMethods.POST)
+                .when(mockClientCredentialsProvider).getHttpMethod();
+
```
Comment (kenmccracken) [inline here-oauth-client/src/test/java/com/here/account/oauth2/HereAccountTest.java]: You could also add a second test that verifies what happens when the clientCredentialProvider's clock is non-null, by adding a second test case that has the additional lines  
```
Clock myClock = Clock.SYSTEM;
Mockito.doReturn(myClock).when(mockClientCredentialsProvider).getClock().
```
